shiftcommerce/shift-react-components
# Feature: Checkout Integration

## Overview

Github Issue: https://github.com/shiftcommerce/shift-front-end-react/issues/843

This PR contains components and styling updates required for the PayPal integration feature.

[PR 1 - Reviewed](https://github.com/shiftcommerce/shift-react-components/pull/2)

It also bumps the package version to v0.5.0

![Payment Method Page](https://user-images.githubusercontent.com/4486874/54922676-8dda5080-4f00-11e9-9450-1eb697c4e13c.png)

Related PRs:
[Shift Next PR](https://github.com/shiftcommerce/shift-next/pull/10)
[Shift Front End React](https://github.com/shiftcommerce/shift-front-end-react/pull/846)

## QA

1. checkout branch `feature/790_checkout_integration`

2. In the `shift-front-end-react` project, checkout branch `issue/790_paypal_checkout_integration` and boot app (assumption is that you are using yalc)

3. Add items to cart and navigate to the first step of checkout - `payment-method` selection

4. Check that the `Continue Shopping` btn in the cart should be full-width

5. Check that the `Payment Method` step is present and highlighted

6. Check that there are payment method options displayed on the page

7. Check that  clicking the `Pay By Credit/Debit Card` button results in the page navigating to checkout step 2 - `Shipping Address`

8. Check that clicking the `PayPal` button in the page navigates to the PayPal website. 

![Payment Method Selection](https://user-images.githubusercontent.com/4486874/54922925-0fca7980-4f01-11e9-8ac0-28a121202637.png)

9. Complete all steps on PayPal and click the continue button

![Shipping Address Selection](https://user-images.githubusercontent.com/4486874/54922936-13f69700-4f01-11e9-9bb5-2b8159d5d604.png)

10. When you have been redirected back to the site, check that the shipping address summary is present and that you are redirect to checkout 3 - shipping method. Also ensure that the `Edit` address button is not present and that clicking the shipping address step in the checkout header steps, redirects you back to checkout 1 - Payment Method selection

![Checkout 3 - Shipping Methods](https://user-images.githubusercontent.com/4486874/54924891-2bd01a00-4f05-11e9-9541-e39d63a8b464.png)
